### ISSUE TITLE
AOT CPP test crash in master

### ISSUE THREAD

**Issue by feisuzhu** [taichi-dev/taichi#6556]
```
...... 
[ RUN      ] CapiTest.AotTestCpuBufferInterop
[W 11/09/22 08:18:51.079 3018] [taichi_core_impl.cpp:ti_set_last_error@130] C-API error: (not supported) taichi::arch_is_cpu(config->arch)
[       OK ] CapiTest.AotTestCpuBufferInterop (214 ms)
[ RUN      ] CapiTest.AotTestCudaBufferInterop
[       OK ] CapiTest.AotTestCudaBufferInterop (468 ms)
[ RUN      ] CapiTest.AotTestVulkanTextureInterop
[I 11/09/22 08:18:51.553 3018] [vulkan_device_creator.cpp:pick_physical_device@399] Found Vulkan Device 0 (NVIDIA GeForce RTX 2060)
[I 11/09/22 08:18:51.553 3018] [vulkan_device_creator.cpp:create_logical_device@469] Vulkan Device "NVIDIA GeForce RTX 2060" supports Vulkan 0 version 1.2.175
[       OK ] CapiTest.AotTestVulkanTextureInterop (35 ms)
[----------] 7 tests from CapiTest (2508 ms total)

[----------] Global test environment tear-down
[==========] 7 tests from 1 test suite ran. (2508 ms total)
[  PASSED  ] 7 tests.
----------------------------------
---- Running Taichi AOT tests ----
----------------------------------
[Taichi] version 1.3.0, llvm 15.0.4, commit 131aacf6, linux, python 3.8.13
[Taichi] Starting on arch=x64
Running main() from /home/dev/taichi/external/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = LlvmAotTest.CpuKernel
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from LlvmAotTest
[ RUN      ] LlvmAotTest.CpuKernel
[       OK ] LlvmAotTest.CpuKernel (252 ms)
[----------] 1 test from LlvmAotTest (252 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (252 ms total)
[  PASSED  ] 1 test.
[Taichi] version 1.3.0, llvm 15.0.4, commit 131aacf6, linux, python 3.8.13
[Taichi] Starting on arch=cuda
Traceback (most recent call last):
  File "tests/run_tests.py", line 396, in <module>
    test()
  File "tests/run_tests.py", line 387, in test
    _test_cpp(args.keys)
  File "tests/run_tests.py", line 129, in _test_cpp
    _test_cpp_aot(cpp_test_filename, build_dir, __aot_test_cases)
  File "tests/run_tests.py", line 53, in _test_cpp_aot
    subprocess.check_call(cmd_list, env=env_copy)
  File "/home/dev/miniconda/envs/py38/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/home/dev/miniconda/envs/py38/bin/python3', '/home/dev/taichi/tests/cpp/aot/python_scripts/kernel_aot_test1.py', '--arch=cuda']' died with <Signals.SIGSEGV: 11>.
```

**Comment by feisuzhu** [comment]
https://github.com/taichi-dev/taichi/actions/runs/3426224873

**Comment by jim19930609** [comment]
Looks like the entire AOT feature is pretty shaky on the current master branch. Let's fix these CI failures before the v1.3.0 release, otherwise I'd be a bit concerned about the release quality, especially in terms of the AOT parts.
